packages/twenty-front/src/modules/workflow/hooks/useAvailableVariablesInWorkflowStep.ts

```diff
@@ -0,0 +1,68 @@
+import { capitalize } from '~/utils/string/capitalize';
+import { useRecoilValue } from 'recoil';
+import { workflowIdState } from '@/workflow/states/workflowIdState';
+import { useWorkflowWithCurrentVersion } from '@/workflow/hooks/useWorkflowWithCurrentVersion';
+import { workflowSelectedNodeState } from '@/workflow/states/workflowSelectedNodeState';
+import { getStepDefinitionOrThrow } from '@/workflow/utils/getStepDefinitionOrThrow';
+import { isDefined } from 'twenty-ui';
+import { StepOutputSchema } from '@/workflow/search-variables/types/StepOutputSchema';
+
+export const useAvailableVariablesInWorkflowStep = (): StepOutputSchema[] => {
```

Comment:
shouldn't we do that logic on backend side? Maybe a custom endpoint that would return the outputschema in the right format for a given step?

Reply:
I did that to reduce backend calls, but I agree there is too much work done by frontend, we may change that in the future